src/api/impl/ConnectedServer.cpp

```diff
@@ -159,11 +159,11 @@ Login::Response ConnectedServer::Login(const Login::Request& request) noexcept
 Logout::Response ConnectedServer::Logout(_UNUSED const Logout::Request& /*request*/) noexcept
 {
     // update the global API pointer to a disconnected API
-    ra::services::ServiceLocator::Provide<ra::api::IServer>(new (std::nothrow) DisconnectedServer(m_sHost));
+    ra::services::ServiceLocator::Provide<ra::api::IServer>(std::make_unique<DisconnectedServer>(m_sHost).release());
```

Comment:
We should probably make a `std::unique_ptr` overload to explicitly show we are transferring ownership of the pointer.

Reply:
The old structure was most similar to constructing a `unique_ptr` instance:
```
unique_ptr<ra::api::IServer> foo(new DisconnectedServer(m_sHost));
```
This seems excessively verbose without any benefit.

I can see that having `Provide` wrap a `unique_ptr` may be confusing, would changing the method name help? `ProvideUnique<T>`?

Reply:
It just warned on using `new` and not deleting it.

I'd think that the `ProvideUnique<T>` should have a unique pointer as a param to show an explicit transfer of ownership, then it releases it's owned pointer to the service container.

Kind of like this:
```
auto pDisconnectedServer = std::make_unique<DisconnectedServer>(m_sHost);
// The param would be by value since it can't be copied
 ra::services::ServiceLocator::Provide<ra::api::IServer>(std::move(pDisconnectedServer ));
```
Inside the `Provide<T>`, the owned pointer would be released.
If you want to call it `ProvideUnique `I guess that's fine.